internal/fuse/file.go

```diff
@@ -123,16 +120,13 @@ func (f *file) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadR
 	}
 
 	// Skip blobs before the offset
-	startContent := 0
-	for offset > int64(f.sizes[startContent]) {
-		offset -= int64(f.sizes[startContent])
-		startContent++
-	}
+	startContent := f.findStart(offset)
```

Comment:
using `import "sort"` (and the changes mentioned above), this can be simply written as:
```
startContent := sort.Search(len(f.cumsize), func(i int) bool {return cumsize[i] > offset})
```

Reply:
That doesn't work (the existing test crashes), because we have to handle two corner cases: size == 0 and size == filesize. The following line is

    offset -= f.cumsize[startContent]

and sort.Search may return 0 through len(f.cumsize), inclusive. This can be handled by special casing, but that's not necessarily more complicated than overallocating and I really liked the one-line summary

    // cumsize[i] holds the cumulative size of blobs[:i].

It's subtle, but the semantics can be described without a +1 or -1 :)